### PR TITLE
Feature/iosas 861

### PR DESCRIPTION
**Pull request by mmumby**
# Main Updates
- Removed all mock data for EOI
- connect EOI form to API
- APIs include: GET all, GET by id, PATCH, POST, PATCH/cancel
- GET metaData for pickLists on form
- POST documents, GET documents
- Updated conditional rendering of fields/auto populating fields based on data

### CODE REVIEW

File: backend/src/components/user.js [hunk 1]
```diff
@@ -13,29 +13,37 @@ async function getEdxUserByDigitalId(accessToken, digitalID, correlationID) {
     const params = {
       params: {
         digitalId: digitalID,
-      }
+      },
     };
-    return await getDataWithParams(accessToken, config.get('edx:edxUsersURL'), params, correlationID);
+    return await getDataWithParams(
+      accessToken,
+      config.get('edx:edxUsersURL'),
```
Comment (fwpushan) [inline backend/src/components/user.js:20]: can we change config name to iosas?

Comment (mmumby) [reply backend/src/components/user.js:20]: Yes - However I just formatted this file, so Im not certain what this is in reference to. I think this entire function will be deleted - Can we address it then?

File: frontend/src/components/EOI/ExpressionOfInterestForm.vue [hunk 2]
```diff
@@ -1,96 +1,306 @@
...
+          <v-row>
+            <v-col cols="12">
+              <v-label class="no-mb"
+                >Is this Expression of Interest for an existing School
+                Authority?</v-label
+              >
+              <v-radio-group
+                id="iosas_existingauthority"
+                v-model="data.iosas_existingauthority"
+                color="#003366"
```
Comment (fwpushan) [inline frontend/src/components/EOI/ExpressionOfInterestForm.vue:46]: Can we manage this with style class only?

File: frontend/src/components/EOI/ExpressionOfInterestForm.vue [hunk 3]
```diff
@@ -1,96 +1,306 @@
...
+                class="mt-4"
+                :rules="[rules.requiredRadio()]"
+                inline
+                @change="validateAndPopulate"
+              >
+                <v-radio label="Yes" color="#003366" :value="true" />
```
Comment (fwpushan) [inline frontend/src/components/EOI/ExpressionOfInterestForm.vue:52]: style class?

File: frontend/src/components/EOI/ExpressionOfInterestForm.vue [hunk 4]
```diff
@@ -1,96 +1,306 @@
 <template>
```
Comment (fwpushan) [inline frontend/src/components/EOI/ExpressionOfInterestForm.vue:1]: This file is very big. Can we refactor them?

File: frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue [hunk 5]
```diff
@@ -1,221 +1,355 @@
...
+      incorporationDocument: null,
+      certificateOfGoodStandingDocument: null,
+      otherDocuments: null,
       NULL_STRING,
     };
   },
+  created() {
+    if (this.eoi?.documents?.length > 0) {
+      this.incorporationDocument = this.eoi.documents.find(
+        ({ iosas_eoidocumenttype }) => iosas_eoidocumenttype === 100000000
```
Comment (fwpushan) [inline frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue:325]: Can we use named/labeled const?  (100000000, 100000001)

File: frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue [hunk 6]
```diff
@@ -1,221 +1,355 @@
...
+      return this.eoi?.iosas_authorityheadfirstname
         ? this.eoi.iosas_authorityheadfirstname +
             ' ' +
-            this.eoi.iosas_authorityheadname
+            this.eoi.iosas_schoolauthorityheadname
         : null;
     },
     getDesignatedHeadName() {
-      return this.eoi.iosas_designatedcontactfirstname
+      return this.eoi?.iosas_designatedcontactfirstname
```
Comment (fwpushan) [inline frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue:352]: is it a js ternary operator? Then what is the return value for the false path?

**Review by fwpushan — COMMENTED** [review]
All good check my comments

File: frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue (hunk 6 above)
Comment (mmumby) [reply frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue:352]: no, it's a null check for the eoi object, if this.eoi is null or undefined, then we can't access the field name.

File: frontend/src/components/EOI/ExpressionOfInterestForm.vue (hunk 2 above)
Comment (mmumby) [reply frontend/src/components/EOI/ExpressionOfInterestForm.vue:46]: I have not found a way to override the vuetify class for the inputs, This color prop changes the active border style all inputs and was the existing pattern on the app.

File: frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue (hunk 5 above)
Comment (mmumby) [reply frontend/src/components/EOI/ExpressionOfInterestReadOnlyView.vue:325]: done!

**Review by fwpushan — APPROVED** [review]

**Pull request merged**